tools/explorer/app.py

```diff
@@ -42,23 +63,59 @@ def main(parsed_args):
       for field in fields:
         print(field)
     elif function_choice == '2':
-      print('This functionality is still under construction.\n' +
-            'Please pick another option.')
+
+      standard_field_list = _InputFields(my_ontology)
+
+      entity_type_match_dict = {}
+      for i, match in enumerate(my_ontology.GetEntityTiypesFromFields(
+          standard_field_list
+      )):
+        entity_type_match_dict[i] = match
+      print('How many matches would you like to see?')
+      match_selection = ast.literal_eval(
+          input('enter a number or 0 for all matches:\n'))
+      if match_selection == 0:
+        for i in entity_type_match_dict.items():
+          print(f'{i+1}. {entity_type_match_dict[i]}')
+      elif match_selection > 0:
+        for i in range(match_selection):
+          print(colored(f'{i+1}. {entity_type_match_dict[i]}', 'green'))
+      print('Would you like to see field comparisons for any of these matches?')
+      visualize = input('(y/n): ')
+      if visualize == 'y':
+        visualize_done = False
+        while not visualize_done:
+          comparison_number = ast.literal_eval(input(
+              'which match would you like to visualize?\nmatch number: '))
+          print_match = entity_type_match_dict[comparison_number - 1]
+          print(my_ontology.PrintFieldSetComparison(print_match))
+          repeat_comparison = input(
+              'Would you like to see another match comparison?\n(y/n): ')
+          if repeat_comparison == 'n':
+            break
+          elif repeat_comparison == 'y':
+            continue
+          else:
+            raise ValueError('Please enter proper value for decision tree')
+      else:
+        continue
+
     elif function_choice == '3':
       namespace = input('Enter a namespace: ').upper()
       field_name = input('Enter a field name to validate: ')
       standard_field = model.StandardField(namespace, field_name)
       field_is_valid = my_ontology.IsFieldValid(standard_field)
-      if not namespace:
+      if namespace == '':
         namespace = 'global'
       print(f'{field_name} is defined in {namespace}: {field_is_valid}')
```

Comment:
please break this into several functions based on the choice.
so you can have three functions based on those: 
        '1: Get fields for a type name\n' +
        '2: Get types for a list of fields\n' +
        '3: Validate a field name\n' +

it is hard to follow and has a lot of nesting.

Reply:
Done.